app/views/session_timeout/_ping.js.erb

```diff
@@ -26,12 +30,12 @@ function success(data) {
       time_cutoff = new Date().getTime() + warning,
       show_warning = time_timeout < time_cutoff;
 
-  if (show_warning & !el) {
-    cntnr.insertAdjacentHTML('afterbegin', warning_info);
+  if (show_warning & !modal.shown) {
```

Comment:
Yes, it works in this instance because we are explicitly comparing booleans, but if we ever change them it could lead to issues.